org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/diagnostics/LSPErrorReporterForXML.java

```diff
@@ -73,7 +74,7 @@ public LSPErrorReporterForXML(DOMDocument xmlDocument, List<Diagnostic> diagnost
 	protected Range toLSPRange(XMLLocator location, String key, Object[] arguments, String message,
 			DiagnosticSeverity diagnosticSeverity, boolean fatalError, DOMDocument document) {
 		String documentURI = location.getExpandedSystemId();

```

Comment:
String grammarURI is probably clearer

Reply:
The URI cannot be the XML document or the grammar.

Perhaps you should prefer 

`String documentOrGrammarURI` ?

Reply:
ok

Reply:
fixed